test/e2e/framework/util.go

```diff
@@ -0,0 +1,154 @@
+	"k8s.io/client-go/1.5/pkg/util/wait"
+	"k8s.io/client-go/1.5/pkg/util/yaml"
+
+	"github.com/coreos/prometheus-operator/pkg/spec"
+)
+
+func CreateDeployment(kclient kubernetes.Interface, ns string, deploy *v1beta1.Deployment) error {
+	if _, err := kclient.Extensions().Deployments(ns).Create(deploy); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func WaitForPodsReady(kclient kubernetes.Interface, ns string, timeout time.Duration, expectedRunning int, opts api.ListOptions) (*v1.PodList, error) {
+	t := time.After(timeout)
+	tick := time.Tick(500 * time.Millisecond)
+
+	for {
+		select {
+		case <-t:
+			return nil, fmt.Errorf("timed out while waiting for pod to be ready")
+		case <-tick:
+			pl, err := kclient.Core().Pods(ns).List(opts)
+			if err != nil {
+				return nil, err
+			}
+
+			running := 0
+			if len(pl.Items) > 0 {
+				for _, p := range pl.Items {
+					if p.Status.Phase == v1.PodRunning {
+						running++
+					}
+				}
+				if running == expectedRunning {
+					return pl, nil
+				}
+			}
+		}
+	}
+}
+
+func PrometheusListOptions(name string) api.ListOptions {
+	return api.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			"app":        "prometheus",
+			"prometheus": name,
+		}),
+	}
+}
+
+func WaitPrometheusTPRReady(httpClient *http.Client, interval, timeout time.Duration, host, ns string) error {
+	return wait.Poll(interval, timeout, func() (bool, error) {
+		resp, err := ListPrometheuses(host, ns, httpClient)
+		if err != nil {
+			return false, err
+		}
+		defer resp.Body.Close()
+
+		switch resp.StatusCode {
+		case http.StatusOK:
+			return true, nil
+		case http.StatusNotFound: // not set up yet. wait.
+			return false, nil
+		default:
+			return false, fmt.Errorf("invalid status code: %v", resp.Status)
+		}
+	})
+}
+
+func ListPrometheuses(host, ns string, httpClient *http.Client) (*http.Response, error) {
+	return httpClient.Get(fmt.Sprintf("%s/apis/monitoring.coreos.com/v1alpha1/namespaces/%s/prometheuses", host, ns))
+}
+
+func MakePromCluster(name string, replicas int) *spec.Prometheus {
+	return &spec.Prometheus{
+		ObjectMeta: v1.ObjectMeta{
+			Name: name,
+		},
+		Spec: spec.PrometheusSpec{
+			Replicas: int32(replicas),
+		},
+	}
+}
+
+func CreatePromCluster(f *Framework, e *spec.Prometheus) (*spec.Prometheus, error) {
+	b, err := json.Marshal(e)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := f.HTTPClient.Post(
+		fmt.Sprintf("%s/apis/monitoring.coreos.com/v1alpha1/namespaces/%s/prometheuses", f.MasterHost, f.Namespace.Name),
+		"application/json", bytes.NewReader(b))
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusCreated {
+		return nil, fmt.Errorf("unexpected status: %v", resp.Status)
+	}
+	decoder := yaml.NewYAMLOrJSONDecoder(resp.Body, 100)
+	res := &spec.Prometheus{}
+	if err := decoder.Decode(res); err != nil {
+		return nil, err
+	}
+	fmt.Printf("created prometheus instance: %v\n", res.Name)
```

Comment:
Fine with providing some form of logging in the e2e tests throughout the code. But we should use proper loggers rather than prints. Should also happen in the framework after a successful call rather than in utility functions.